src/auth.rs

```diff
@@ -94,21 +94,23 @@ impl<const USER_ROLE: u32> FromRequestParts<AppState> for AuthExtractor<USER_ROL
             remove_cookie_on_fail = remove_cookie_on_fail.domain("localhost");
         }
 
-        let session_id = cookies
-            .private(key)
-            .get(SESSION_COOKIE_NAME)
-            .ok_or_else(|| {
+        let session_id = match cookies.private(key).get(SESSION_COOKIE_NAME) {
+            None => {
                 tracing::error!("auth-extractor: failed to get session_id cookie");
-                cookies.remove(remove_cookie_on_fail.clone().finish());
-                AuthError::InvalidSession
-            })?;
+                cookies.remove(remove_cookie_on_fail.finish());
+                return Err(AuthError::InvalidSession);
+            }
+            Some(id) => id,
```

Comment:
Indentation here is a little weird

Reply:
what's weird about it? I ran `cargo fmt` on it and it produced the same formatting.

Reply:
Will check it on my laptop